server/text_generation_server/models/custom_modeling/flash_llama_modeling.py

```diff
@@ -212,34 +240,54 @@ def __init__(self, prefix, config, weights):
         # Fuse gate and up proj
         bias = getattr(config, "mlp_bias", False)
         if config.model_type == "phi3":
-            self.gate_up_proj = TensorParallelColumnLinear.load_gate_up(
+            gate_up_proj = TensorParallelColumnLinear.load_gate_up(
                 config,
                 prefix=f"{prefix}.gate_up_proj",
                 weights=weights,
                 bias=bias,
             )
         else:
-            self.gate_up_proj = TensorParallelColumnLinear.load_multi(
+            gate_up_proj = TensorParallelColumnLinear.load_multi(
                 config,
                 prefixes=[f"{prefix}.gate_proj", f"{prefix}.up_proj"],
                 weights=weights,
                 dim=0,
                 bias=bias,
             )
-        self.down_proj = TensorParallelRowLinear.load(
+
+        self.gate_up_proj = TensorParallelMultiAdapterLinear.load(
+            gate_up_proj,
+            index,
+            ["gate_proj", "up_proj"],
```

Comment:
Are the LoRa parameters also separate in the Phi3 case, since Phi3 uses a single gate+up projection?

Reply:
great catch, since the `adapter_target_to_layer` is not implemented for flash_phi I believe this could cause and issue. 

I've updated the PR to avoid using `TensorParallelMultiAdapterLinear` in any special case (phi3 and baichuan)